### DISCUSSION TITLE
Unfamiliar errors running methylKit on mox

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1213] [Q&A]
## TL;DR R module on mox producing new errors when running randomization tests

I'm running some pretty computationally-intensive `methylKit` randomization tests. This is my [R script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/06-methylKit-randtest.R), and this is the [SLURM script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/06-methylKit-SLURM.sh) I'm using to load the R module and run the R script.

When running this code, I encountered the errors below:

```
for (i in 1:1000) {
  pHRandTreat <- sample(sampleMetadataFem$pHTreatment,
                        size = length(sampleMetadataFem$pHTreatment),
                        replace = FALSE)
  pHRandDML <- methylKit::reorganize(methylationInformationFilteredCov5Fem,
                                     sample.ids = sampleMetadataFem$sampleID,
                                     treatment = pHRandTreat) %>%
    methylKit::calculateDiffMeth(., covariates = covariateMetadataFem,
                                 overdispersion = "MN",
                                 test = "Chisq", mc.cores = 40) %>%
    methylKit::getMethylDiff(., difference = 75, qvalue = 0.01) %>%
    nrow() -> pHRandTest75Fem[i]
}
```

![Screen Shot 2021-05-17 at 5 48 46 PM](https://user-images.githubusercontent.com/22335838/118573985-6aa23900-b738-11eb-9519-139184e4ed61.png)

I haven't encountered the "non-existent physical address" or cores not delivering results errors previously. I did try running this code on the R Studio server without the `mc.cores` argument. However, it took 2 days to make it through 6 iterations, and when I included the `mc.cores` argument, the R Studio server browser interface froze (see notebook [here](https://yaaminiv.github.io/WGBS-Analysis-Part24/)). 

## What do these errors mean, and how do I not get them?

**Comment by yaaminiv** [comment]
I'm currently running the randomization test WITHOUT `mc.cores` on mox to see if I still encounter the "non-existent physical address" errors and will update this discussion accordingly.

**Reply by yaaminiv** [reply]
Running without `mc.cores` and I have yet to encounter the same error. For reference, I got the previous error ~10 minutes after I started running the script. My guess is that `mc.cores` is the issue.

Should I try running with `mc.cores = 28`, since we're usually allowed 28 cores? When I tried running without `mc.cores` on the R Studio server, I only made it through 6 iterations in 2 days, and I need to get through 1000 iterations...
